include/odp/api/spec/dma_types.h

```diff
@@ -492,6 +536,77 @@ typedef struct odp_dma_transfer_param_t {
 	 */
 	odp_dma_seg_t *dst_seg;
 
+	/** Destination segment pool
+	 *
+	 *  When odp_dma_transfer_param_t::dst_alloc is set, provide the pool details from which
+	 *  the buffers need to be allocated.
+	 */
+	odp_pool_t dst_seg_pool;
+
+	/** Transfer options
+	 *
+	 *  Options to further configure the transfer.
+	 *  Initialize all unused bits to zero.
+	 */
+	union {
+		/** Option bit fields */
+		struct {
+			/** Allow freeing all the source segments
+			 *
+			 *  When set to 1, all source segments with data format
+			 *  #ODP_DMA_FORMAT_PACKET are freed.
```

Comment:
Just realized that something probably needs to be said about in which cases the freeing is carried out. E.g. if transfer has failed, then probably the freeing is not done by implementation but application needs to do it. 

So could be amended:
```
*  When set to 1, all source segments with data format
*  #ODP_DMA_FORMAT_PACKET are freed in successful
*  transfers.
```